packages/ibm-products/src/components/ConditionBuilder/ConditionGroupBuilder/ConditionGroupBuilder.js

```diff
@@ -33,29 +35,52 @@ const ConditionGroupBuilder = ({
     }
   };
 
-  const onChangeHandler = (updatedConditions, conditionIndex) => {
+  const onChangeHandler = (updatedCondition, conditionId) => {
     onChange({
-      ...state,
-      conditions: state.conditions.map((condition, cIndex) =>
-        conditionIndex === cIndex ? updatedConditions : condition
+      ...group,
+      conditions: group.conditions.map((condition) =>
+        condition.id == conditionId ? updatedCondition : condition
       ),
     });
   };
 
   const addConditionHandler = (conditionIndex) => {
+    let newCondition = {
+      property: undefined,
+      operator: '',
+      value: '',
+      popoverToOpen: 'propertyField',
+      id: uuidv4(),
+    };
+
+    let currentConditions = JSON.parse(JSON.stringify(group.conditions));
+    currentConditions.splice(conditionIndex + 1, 0, newCondition);
     onChange({
-      ...state,
-      conditions: [
-        ...state.conditions.slice(0, conditionIndex + 1),
-        {
-          property: undefined,
-          operator: '',
-          value: '',
-          popoverToOpen: 'propertyField',
-        },
-        ...state.conditions.slice(conditionIndex + 1),
-      ],
+      ...group,
```

Comment:
you had it correct before! using the similar approach as mentioned
```
    const newIdx = conditionIndex + 1;

    const newCondition = {
      property: undefined,
      operator: '',
      value: '',
      popoverToOpen: 'propertyField',
      id: uuidv4(),
    };

    onChange({
      ...group,
      conditions: [
        ...group.conditions.slice(0, newIdx),
        newCondition,
        ...group.conditions.slice(newIdx + 1),
      ],
    });
```

Reply:
modified by using slice()